src/pathfinding_service/model/channel_view.py

```diff
@@ -16,6 +17,60 @@
 from raiden_libs.marshmallow import ChecksumAddress
 
 
+class Interpolate:  # pylint: disable=too-few-public-methods
+    def __init__(self, x_list: Sequence, y_list: Sequence):
+        if any(y - x <= 0 for x, y in zip(x_list, x_list[1:])):
+            raise ValueError("x_list must be in strictly ascending order!")
+        self.x_list = x_list
+        self.y_list = y_list
+        intervals = zip(x_list, x_list[1:], y_list, y_list[1:])
+        self.slopes = [(y2 - y1) / (x2 - x1) for x1, x2, y1, y2 in intervals]
+
+    def __call__(self, x: float) -> float:
+        if not self.x_list[0] <= x <= self.x_list[-1]:
+            raise ValueError("x out of bounds!")
+        if x == self.x_list[-1]:
+            return self.y_list[-1]
+        i = bisect_right(self.x_list, x) - 1
+        return self.y_list[i] + self.slopes[i] * (x - self.x_list[i])
+
+
+@dataclass
+class FeeSchedule:
+    # pylint: disable=not-an-iterable
+    flat: FeeAmount = FeeAmount(0)
+    proportional: float = FeeAmount(0)
+    imbalance_penalty: Optional[List[List[TokenAmount]]] = None
+    _penalty_func: Interpolate = field(init=False, repr=False)
+
+    def __post_init__(self) -> None:
+        if self.imbalance_penalty:
+            assert isinstance(self.imbalance_penalty, list)
+            x_list, y_list = tuple(zip(*self.imbalance_penalty))
+            self._penalty_func = Interpolate(x_list, y_list)
+
+    def fee(self, amount: TokenAmount, capacity: TokenAmount) -> FeeAmount:
+        if self.imbalance_penalty:
+            # Total channel capacity - node capacity = balance (used as x-axis for the penalty)
+            balance = self._penalty_func.x_list[-1] - capacity
+            imbalance_fee = self._penalty_func(balance + amount) - self._penalty_func(balance)
+        else:
+            imbalance_fee = 0
+        return FeeAmount(round(self.flat + amount * self.proportional + imbalance_fee))
+
+    def reversed(self) -> "FeeSchedule":
+        if not self.imbalance_penalty:
```

Comment:
Should we check for explicitely for `None`  here? Not sure what would happen if the client would send an empty list!?

Reply:
For empty lists we could either ignore them (as done now) or throw an error. Since it is not a useful input, I would be fine with either.